packages/fxa-auth-server/lib/routes/validators.js

```diff
@@ -384,25 +388,63 @@ module.exports.subscriptionsSubscriptionListValidator = isA.object({
 // - metadata can contain arbitrary keys that we don't expect (e.g. used by other systems)
 // - but we can make a good effort at validating what we expect to see when we see it
 module.exports.subscriptionPlanMetadataValidator = isA.object().unknown(true);
-module.exports.subscriptionProductMetadataValidator = isA
+
+const capabilitiesClientIdPattern = /^capabilities/;
+const legalResourceDomainPattern =
+  /^https:\/\/accounts-static\.cdn\.mozilla\.net\/legal\/(.*)/;
+
+module.exports.subscriptionProductMetadataBaseValidator = isA
   .object({
+    webIconURL: isA.string().uri().required(),
+    upgradeCTA: isA.string().optional(),
+    downloadURL: isA.string().uri().required(),
+    appStoreLink: isA.string().uri().optional(),
+    playStoreLink: isA.string().uri().optional(),
     productSet: isA.string().optional(),
     productOrder: isA.number().optional(),
-    iconURL: isA.string().optional(),
-    upgradeCTA: isA.string().optional(),
-    downloadURL: isA.string().optional(),
-    appStoreLink: isA.string().optional(),
-    playStoreLink: isA.string().optional(),
+    'product:termsOfServiceDownloadURL': isA
+      .string()
+      .regex(legalResourceDomainPattern)
+      .required(),
+    'product:termsOfServiceURL': isA.string().uri().required(),
+    'product:privacyNoticeDownloadURL': isA
+      .string()
+      .regex(legalResourceDomainPattern)
+      .required(),
+    'product:privacyNoticeURL': isA.string().uri().required(),
+  })
+  .pattern(capabilitiesClientIdPattern, isA.string(), {
+    fallthrough: true,
   })
   .unknown(true);
 
+module.exports.subscriptionProductMetadataValidator = {
+  validate: function (metadata) {
+    const capabilitiesRegex = new RegExp(capabilitiesClientIdPattern, 'g');
```

Comment:
`capabilitiesClientIdPattern` is already a RegExp so there's no need to recreate it here.